### DISCUSSION TITLE
Gist option: Push svg image to a gist

### DISCUSSION THREAD

**Discussion by luisfalconmx** [lowlighter/metrics#207] [Ideas]
I would like the option to generate the `github-metrics.svg` file added, but save it to a github gist and not the repository.

Personally, I don't like Github Actions making changes and commits to the repository because it fills up with contributions that I haven't made myself.

This would be an example of the file that would have to be generated with Github Actions but saved in a gist. (I uploaded the svg file manually, there is still no code created)

- Gist: https://gist.github.com/luisfalconmx/6b4841f554d7168f1111029976108777
- SVG to insert in README.md: https://gist.githubusercontent.com/luisfalconmx/6b4841f554d7168f1111029976108777/raw/github-metrics.svg

I would love to know what you think or if this is possible.

**Comment by lowlighter** [comment]
Committer is the Github Actions bot so its contributions are normally not counted in your own.

I think storing metrics in gists would be a nice feature to integrate though, especially for projects using the repository template 👍🏼
From what I checked quickly, updating gists content can be done with [`octokit.rest.gists.update`](https://octokit.github.io/rest.js/v18#gists-update).

Handling gists creation in metrics seems too complicated because we would need to:
* Find a reliable way to link a job/workflow with a single gist (users may want to use multiple gists or a single one depending on  repositories/workflows)
* Check whether gist has already be created or not (and use [`octokit.rest.gists.create`](https://octokit.github.io/rest.js/v18#gists-create) or [`octokit.rest.gists.update`](https://octokit.github.io/rest.js/v18#gists-update) accordingly)
* Handle users with more than 100 gists because of [`octokit.rest.gists.list`](https://octokit.github.io/rest.js/v18#gists-list) limitations

So I think the best is to go on a simpler API where metrics only manage gists updates:
* Users add add `gists` scope to their token, create a new gist, and pass its `gist_id` to GitHub actions
* Metrics retrieve `gist_id` from user inputs and use [`octokit.rest.gists.update`](https://octokit.github.io/rest.js/v18#gists-update) to update metrics 
 
How does it sounds 🙂 ?

**Comment by lowlighter** [comment]
This has been implemented in #212 

To use this feature, add the following to your workflow:
```yaml
- uses: lowlighter/metrics@master # will be available on @latest when v3.7 releases
  with:
    # other options...
    committer_gist: 12e52345cde46b891264e376f0719166 # your gist id as shown in gist url
    output_action: gist
```

Because `${{ github.token }}` is scoped to repository, it'll use your personal token instead to upload files into your gists, which means you must grant the `gists` scope to your token in order to make it work.

Let me know if you have trouble to set it up 👍🏼 

Thanks for using metrics 🙂 !
